mm2src/coins/utxo.rs

```diff
@@ -2647,5 +2662,100 @@ fn kmd_interest(height: Option<u64>, value: u64, lock_time: u64, current_time: u
     }
     // next 2 lines ported as is from Komodo codebase
     minutes -= 59;
-    (value / 10_512_000) * minutes
+    let accrued = (value / 10_512_000) * minutes;
+
+    Ok(accrued)
+}
+
+fn kmd_interest_accrue_stop_at(height: u64, lock_time: u64) -> u64 {
+    let minutes = if height < 1_000_000 {
+        // interest stop accruing after 1 year before block 1000000
+        365 * 24 * 60
+    } else {
+        // interest stop accruing after 1 month past 1000000 block
+        31 * 24 * 60
+    };
+
+    lock_time + minutes
+}
+
+#[derive(Debug, Serialize, Eq, PartialEq)]
+enum KmdRewardsNotAccruedReason {
+    LocktimeNotSet,
+    LocktimeLessThanThreshold,
+    UtxoHeightGreaterThanEndOfEra,
+    UtxoAmountLessThanTen,
+    OneHourNotPassedYet,
+    TransactionInMempool,
+}
+
+#[derive(Serialize)]
+enum KmdRewardsAccrueInfo {
+    Accrued(BigDecimal),
+    NotAccruedReason(KmdRewardsNotAccruedReason),
+}
+
+#[derive(Serialize)]
+pub struct KmdRewardsInfoElement {
+    tx_hash: H256Json,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    height: Option<u64>,
+    /// The zero-based index of the output in the transaction’s list of outputs.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    input_index: Option<u32>,
+    amount: BigDecimal,
+    locktime: u64,
+    /// Amount of accrued rewards.
+    accrued_rewards: KmdRewardsAccrueInfo,
+    /// Rewards accruing stop at this time for the given transaction.
+    /// None if the transaction is not mined yet.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    accrue_stop_at: Option<u64>,
+}
+
+/// Get rewards info of unspent outputs.
+/// The list is ordered by the output value.
+pub async fn kmd_rewards_info(coin: &UtxoCoin) -> Result<Vec<KmdRewardsInfoElement>, String> {
+    if coin.ticker != "KMD" {
+        return ERR!("rewards info can be obtained for KMD only");
+    }
+    let mut result = Vec::new();
+
+    let rpc_client = coin.rpc_client();
+    let unspents = try_s!(rpc_client.list_unspent_ordered(&coin.my_address).compat().await);
+    for unspent in unspents {
+        let tx_hash: H256Json = unspent.outpoint.hash.reversed().into();
+        let tx_info = try_s!(rpc_client.get_verbose_transaction(tx_hash.clone()).compat().await);
+
+        let value = unspent.value;
+        let locktime = tx_info.locktime as u64;
+        let current_time = try_s!(coin.get_current_mtp().await) as u64;
+        let accrued_rewards = match kmd_interest(tx_info.height, value, locktime, current_time) {
+            Ok(interest) => KmdRewardsAccrueInfo::Accrued(big_decimal_from_sat(interest as i64, coin.decimals)),
+            Err(reason) => KmdRewardsAccrueInfo::NotAccruedReason(reason),
+        };
+        let accrue_stop_at = tx_info
+            .height
+            .clone()
+            .map(|height| kmd_interest_accrue_stop_at(height, locktime));
+        let input_index = match tx_info.height {
```

Comment:
What is the reason to make input_index optional matching on tx height? It doesn't depend on whether transaction is mined.
I've also realized that it might be more clear to rename this field to `outpoint_index` or `output_index` because it represents the position of the unspent *output* in transaction.

Reply:
I little confused with the inpiut_index field. Thanks!